File: Week 2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>



// Declaring functions
int count_letters(string);
int count_words(string);
int count_sentences(string);


int main(void)
{
    // Getting user input
    string text = get_string("Text: ");

    // Counting letters
    int letters = count_letters(text);

    // counting words
    int words = count_words(text);

    // counting sentences
    int sentences = count_sentences(text);


    // Putting it all together
    float L = ((float) letters / (float) words) * 100;
    float S = ((float) sentences / (float) words) * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;

    // Rounding index to make it a grade
    int grade = (int) round((double)index);

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}


// Functions
int count_letters(string text)
{
    int letters = 0;
    for (int i = 0, l = strlen(text); i < l; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}


int count_words(string text)
{
    int words = 1;
    for (int i = 0, l = strlen(text); i < l; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }
    return words;
}



int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0, l = strlen(text); i < l; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

File: Week 6/sentimental-readability/readability.py
def main():
    # Get user input
    text = input("Text: ")

    # Counting letters
    letters = count_letters(text)

    # Counting words
    words = count_words(text)

    # Counting sentences
    sentences = count_sentences(text)

    # Putting it all together
    L = (letters / words) * 100
    S = (sentences / words) * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    grade = round(index)

    if grade < 1:
        print("Before Grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


# Function to count letters
def count_letters(text):
    counter = 0
    for c in text:
        if c.isalpha():
            counter += 1
    return counter


# Function to count words
def count_words(text):
    counter = 1
    for c in text:
        if c == " ":
            counter += 1
    return counter


# Function to count sentences
def count_sentences(text):
    counter = 0
    for c in text:
        if c == "." or c == "!" or c == "?":
            counter += 1
    return counter


main()